### PR TITLE
added view for 404 and 500 exception

### PR DESCRIPTION
**Pull request by kuranovandry**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -27,6 +27,7 @@ gem 'angularjs-rails'
 gem 'angular-ui-bootstrap-rails'
 gem 'angularjs-rails-resource', '~> 2.0.0'
 gem 'sass-rails', '>= 3.2'
+gem 'newrelic_rpm'
```
Comment (maksymst) [inline Gemfile]: Why do you need this gem?

File: config/initializers/unicorn_new_relic_support.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Ensure the agent is started using Unicorn.
+# This is needed when using Unicorn and preload_app is not set to true.
+# See https://docs.newrelic.com/docs/ruby/no-data-with-unicorn
+if defined?(Unicorn) && File.basename($0).start_with?('unicorn')
```
Comment (maksymst) [inline config/initializers/unicorn_new_relic_support.rb]: Why do you need this initializer?

File: app/assets/javascripts/angular/controller/users_controller.coffee [hunk 3]
```diff
@@ -58,7 +98,7 @@ myApp.controller 'UserUpdateController', [
           $location.path '/'
           return
         ), (error) ->
-          console.log error
+          $location.path '/500'
```
Comment (maksymst) [inline app/assets/javascripts/angular/controller/users_controller.coffee:97]: you have Interceptor for this purpose

File: app/assets/javascripts/angular/controller/users_controller.coffee [hunk 4]
```diff
@@ -77,7 +117,7 @@ myApp.controller 'UserAddController', [
...
```
Comment (maksymst) [inline app/assets/javascripts/angular/controller/users_controller.coffee:116]: you have Interceptor for this purpose

File: app/assets/javascripts/angular/controller/users_controller.coffee [hunk 5]
```diff
@@ -23,6 +23,47 @@ myApp.factory 'User', [
...
+]
+myApp.factory('serverErrorInterceptor', [
+  '$rootScope', '$q', '$location'
+  ($rootScope, $q, $location) ->
+    return {
+      responseError: (response) ->
+        if response.status is 500
+          deferred = $q.defer()
+          $location.path('/500')
+          $rootScope.$broadcast 'data:loaded',
```
Comment (maksymst) [inline app/assets/javascripts/angular/controller/users_controller.coffee]: why do you need this?

File: app/assets/javascripts/angular/controller/users_controller.coffee [hunk 6]
```diff
@@ -23,6 +23,47 @@ myApp.factory 'User', [
...
 ]
+myApp.factory('routingErrorInterceptor', [
+  '$rootScope', '$q', '$location'
+  ($rootScope, $q, $location) ->
+    return {
+      responseError: (response) ->
+        if response.status is 404
+          deferred = $q.defer()
+          $location.path('/404')
+          $rootScope.$broadcast 'data:loaded',
```
Comment (maksymst) [inline app/assets/javascripts/angular/controller/users_controller.coffee]: why do you need this?

File: app/assets/javascripts/angular/controller/users_controller.coffee (hunk 3 above)
Comment (kuranovandry) [inline app/assets/javascripts/angular/controller/users_controller.coffee:97]: after adding Interceptor forgot to remove

File: app/assets/javascripts/angular/controller/users_controller.coffee (hunk 4 above)
Comment (kuranovandry) [inline app/assets/javascripts/angular/controller/users_controller.coffee:116]: after adding Interceptor forgot to remove

File: app/assets/javascripts/angular/controller/users_controller.coffee (hunk 5 above)
Comment (kuranovandry) [inline app/assets/javascripts/angular/controller/users_controller.coffee]: He catches the error from the server

File: app/assets/javascripts/angular/controller/users_controller.coffee (hunk 6 above)
Comment (kuranovandry) [inline app/assets/javascripts/angular/controller/users_controller.coffee]: He catches the error from the server

File: app/assets/javascripts/angular/controller/users_controller.coffee (hunk 5 above)
Comment (maksymst) [inline app/assets/javascripts/angular/controller/users_controller.coffee]: $rootScope.$broadcast 'data:loaded' does not catch anything

File: app/assets/javascripts/angular/controller/users_controller.coffee (hunk 6 above)
Comment (maksymst) [inline app/assets/javascripts/angular/controller/users_controller.coffee]: $rootScope.$broadcast 'data:loaded' does not catch anything

File: app/assets/javascripts/angular/controller/users_controller.coffee (hunk 5 above)
Comment (kuranovandry) [inline app/assets/javascripts/angular/controller/users_controller.coffee]: Yes it's my mistake.

**Pull request merged**